tests/test_device.py

```diff
@@ -67,6 +68,27 @@ def test_create_device_with_unsupported_mpi_buf_size():
         pass
 
 
+@pytest.mark.skipif(
+    device_name != "lightning.gpu",
+    reason="Check if the method is pickleable throught the cpp layer",
+)
+def test_devpool_is_pickleable():
+    dev = qml.device(device_name, wires=2)
+    try:
+        pickled_devpool = pkl.dumps(dev._dp)
+        un_pickled_devpool = pkl.loads(pickled_devpool)
+
+        from pennylane_lightning.lightning_gpu_ops import DevPool
+
+        d = DevPool()
+
+        assert isinstance(un_pickled_devpool, DevPool)
+        assert un_pickled_devpool.getTotalDevices() == d.getTotalDevices()
+
+    except Exception:
```

Comment:
Any exception? Could we more precise about the type of exception here?

Reply:
The original error being triggered was a TypeError, but I used any exception because there could be other reasons it fails along the way. I can replace this with a TypeError to be specific.

Reply:
I'm generally in favour of specifying the error type so that we will find out sooner in future if some unexpected different error occurs. I let you decide here. 
```suggestion
    except TypeError:
```